packages/core/content-type-builder/admin/src/components/Relation/RelationField/RelationTargetPicker/index.js

```diff
@@ -5,18 +5,15 @@ import get from 'lodash/get';
 import { MenuItem, SimpleMenu } from '@strapi/design-system/SimpleMenu';
 import useDataManager from '../../../../hooks/useDataManager';
 import { ON_CHANGE_RELATION_TARGET } from '../../../FormModal/constants';
+import { isAllowedContentTypesForRelations } from '../../../../utils';
 
 const RelationTargetPicker = ({ oneThatIsCreatingARelationWithAnother, target }) => {
   const { contentTypes, sortedContentTypesList } = useDataManager();
   const dispatch = useDispatch();
   // TODO: replace with an obj { relation: 'x', bidirctional: true|false }
-  const allowedContentTypesForRelation = sortedContentTypesList
-    .filter(obj => obj.kind === 'collectionType' && obj.visible)
-    .filter(
-      obj =>
-        obj.restrictRelationsTo === null ||
-        (Array.isArray(obj.restrictRelationsTo) && obj.restrictRelationsTo.length > 0)
-    );
+  const allowedContentTypesForRelation = sortedContentTypesList.filter(
+    isAllowedContentTypesForRelations
```

Comment:
FYI A general reco when using functions in filter is to do `.filter(value => function(value))` to avoid eventual issue if you add a second arguments in `isAllowedContentTypesForRelations`.